applications/GeoMechanicsApplication/custom_workflows/dgeoflow.h

```diff
@@ -46,56 +46,109 @@
-        typedef typename MixedGenericCriteriaType::ConvergenceVariableListType ConvergenceVariableListType;
+        using ConvergenceCriteriaType = ConvergenceCriteria<SparseSpaceType, LocalSpaceType>;
+        using MixedGenericCriteriaType = MixedGenericCriteria<SparseSpaceType, LocalSpaceType>;
+        using ConvergenceVariableListType = typename MixedGenericCriteriaType::ConvergenceVariableListType;
 
-        typedef GeoMechanicsNewtonRaphsonErosionProcessStrategy<SparseSpaceType, LocalSpaceType, LinearSolverType>
-            GeoMechanicsNewtonRaphsonErosionProcessStrategyType;
+        using GeoMechanicsNewtonRaphsonErosionProcessStrategyType =
+                GeoMechanicsNewtonRaphsonErosionProcessStrategy<SparseSpaceType, LocalSpaceType, LinearSolverType>;
 
         // Dof arrays
-        typedef SetIdentityFunction<Dof<double>> result_type;
-        typedef PointerVectorSet<Dof<double>, SetIdentityFunction<Dof<double>>, std::less<result_type>,
-                                 std::equal_to<result_type>, Dof<double> *>
-            DofsArrayType;
+        using result_type = SetIdentityFunction<Dof<double>>;
+
 
         static ConvergenceCriteriaType::Pointer setup_criteria_dgeoflow();
         static LinearSolverType::Pointer setup_solver_dgeoflow();
         static GeoMechanicsNewtonRaphsonErosionProcessStrategyType::Pointer setup_strategy_dgeoflow(ModelPart &model_part);
-        static std::vector<std::shared_ptr<Process>> parseProcess(ModelPart &model_part, Parameters projFile);
+        void parseProcess(ModelPart &model_part, Parameters projFile);
+
+
+        struct CriticalHeadInfo
+        {
+            double minCriticalHead = 0.0;
+            double maxCriticalHead = 0.0;
+            double stepCriticalHead = 0.0;
+
+            CriticalHeadInfo(double minCriticalHead, double maxCriticalHead, double stepCriticalHead) :
+                    minCriticalHead(minCriticalHead), maxCriticalHead(maxCriticalHead), stepCriticalHead(stepCriticalHead)
+            {}
+        };
+
+        struct CallBackFunctions
+        {
+            std::function<void(const char*)> rLogCallback;
+            std::function<void(double)> rReportProgress;
+            std::function<void(const char*)> rReportTextualProgress;
+            std::function<bool()> rShouldCancel;
+
+            CallBackFunctions(std::function<void(const char*)> rLogCallback,
+                              std::function<void(double)> rReportProgress,
+                              std::function<void(const char*)> rReportTextualProgress,
+                              std::function<bool()> rShouldCancel) :
+                    rLogCallback(std::move(rLogCallback)), rReportProgress(std::move(rReportProgress)), rReportTextualProgress(std::move(rReportTextualProgress)),
+                    rShouldCancel(std::move(rShouldCancel))
+            {}
+        };
+
+        int ExecuteFlowAnalysis(std::string_view rWorkingDirectory,
```

Comment:
Again a minor style thing (the parameter is now passed by value and no longer by reference):
```suggestion
        int ExecuteFlowAnalysis(std::string_view WorkingDirectory,
```